src/winstanley/wdl.bnf

```diff
@@ -28,7 +28,9 @@ call_input ::= INPUT COLON (mapping (COMMA mapping)*)?
 mapping ::= IDENTIFIER EQUAL expression
 alias ::= AS IDENTIFIER
 wf_outputs ::= OUTPUT LBRACE wf_output* RBRACE
-wf_output ::= fqn wf_output_wildcard?
+wf_output ::= wf_output_declaration | wf_output_wildcard_statement
+wf_output_declaration ::= type_e IDENTIFIER EQUAL expression
+wf_output_wildcard_statement ::= fqn wf_output_wildcard?
```

Comment:
So I tried but couldn't come up with a way to have it accept one or the other but not both.
What you suggested doesn't work because `wf_output_declaration*` matches the empty string so it never goes to the `wf_output_wildcard_statement` branch.
I tried `wf_outputs ::= OUTPUT LBRACE (wf_output_declaration+ | wf_output_wildcard_statement+)* RBRACE` to but it also accepts both :/